rest_framework/permissions.py

```diff
@@ -108,8 +108,21 @@ class IsAuthenticated(BasePermission):
     """
     Allows access only to authenticated users.
     """
+    # DRF should authorize all OPTIONS requests by default #5616
+    def has_permission(self, request, view):
+        if request.method == 'OPTIONS':
+            return True
+        return request.user and request.user.is_authenticated
```

Comment:
`IsAuthenticated` and `IsAuthenticatedOrOptionsOnly` are identical. I assume these changes were supposed to be removed given the additional of `IsAuthenticatedOrOptionsOnly`.